dbt/include/sqlserver/macros/materializations/models/table/create_table_as.sql

```diff
@@ -1,26 +1,35 @@
 {% macro sqlserver__create_table_as(temporary, relation, sql) -%}
+   {#- TODO: add contracts here when in dbt 1.5 -#}
+   {%- set sql_header = config.get('sql_header', none) -%}
    {%- set as_columnstore = config.get('as_columnstore', default=true) -%}
-   {% set tmp_relation = relation.incorporate(
-   path={"identifier": relation.identifier.replace("#", "") ~ '_temp_view'},
-   type='view')-%}
    {%- set temp_view_sql = sql.replace("'", "''") -%}
+   {%- set tmp_relation = relation.incorporate(
```

Comment:
if we're using temp tables correctly, shouldn't this be the below?
when is `.replace("#", "")` necessary?
also 
```suggestion
 {% set tmp_relation = make_temp_relation(relation,'_temp_view')
```

Reply:
This is doing it the other way around. create_table_as is called to create temporary tables when working with snapshots and incremental models. In that case the `temporary` argument is set to `true`. In Postgres, this just means you have to add the keyword `temporary` in your CREATE TABLE syntax, but in SQL Server we have to prefix with `#`. This happens in `sqlserver__make_temp_relation` in relation.sql. In that case we will receive a relation identifier here which already has the `#` prefix. But for our temp view (see other comment below), we cannot have the `#` prefix in there. So that's removed here so that we can continue with business as usual.